src/CloudNative.CloudEvents.Amqp/AmqpClientExtensions.cs

```diff
@@ -94,5 +95,96 @@ private static bool HasCloudEventsContentType(Message message, out string conten
             return contentType?.StartsWith(CloudEvent.MediaType) == true;
         }
+
+        public static Message ToAmqpMessage(this CloudEvent cloudEvent, ContentMode contentMode, CloudEventFormatter formatter)
+        {
+            var applicationProperties = MapHeaders(cloudEvent);
+            RestrictedDescribed bodySection;
+            Properties properties;
+
+            switch (contentMode)
+            {
+                case ContentMode.Structured:
+                    bodySection = new Data
+                    {
+                        Binary = formatter.EncodeStructuredModeMessage(cloudEvent, out var contentType)
+                    };
+                    // TODO: What about the other parts of the content type?
+                    properties = new Properties { ContentType = contentType.MediaType };
+                    break;
+                case ContentMode.Binary:
+                    bodySection = SerializeData(cloudEvent.Data);
+                    properties = new Properties { ContentType = cloudEvent.DataContentType };
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(contentMode), $"Unsupported content mode: {contentMode}");
+            }
+            return new Message
+            {
+                ApplicationProperties = applicationProperties,
+                BodySection = bodySection,
+                Properties = properties
+            };
+        }
+
+        private static ApplicationProperties MapHeaders(CloudEvent cloudEvent)
+        {
+            var applicationProperties = new ApplicationProperties();
+            var properties = applicationProperties.Map;
+            properties.Add(SpecVersionAmqpHeader, cloudEvent.SpecVersion.VersionId);
+
+            foreach (var pair in cloudEvent.GetPopulatedAttributes())
+            {
+                var attribute = pair.Key;
+
+                // The content type is specified elsewhere.
+                if (attribute == cloudEvent.SpecVersion.DataContentTypeAttribute)
+                {
+                    continue;
+                }
+
+                string propKey = AmqpHeaderPrefix + attribute.Name;
+
+                // TODO: Check that AMQP can handle byte[], bool and int values
+                object propValue = pair.Value switch
+                {
+                    Uri uri => uri.ToString(),
+                    // AMQPNetLite doesn't support DateTimeOffset values, so convert to UTC.
+                    // That means we can't roundtrip events with non-UTC timestamps, but that's not awful.
+                    DateTimeOffset dto => dto.UtcDateTime,
+                    _ => pair.Value
+                };
+                properties.Add(propKey, propValue);
+            }
+            return applicationProperties;
+        }
+
+        /// <summary>
+        /// Convert data into a suitable format for inclusion in an Avro record.
```

Comment:
Looks like this was copied incorrectly as it references Avro.

Reply:
It was copied accurately from the now-deleted AmqpCloudEventMessage - which of course was broken :)
Fixed now.